lib/matplotlib/axes/_axes.py

```diff
@@ -1241,6 +1245,7 @@ def eventplot(self, positions, orientation='horizontal', lineoffsets=1,
         return colls
 
     #### Basic plotting
+    @unpack_labeled_data()
```

Comment:
This function is going to need to be special cased, it does not plot anything if you pass data in as kwargs. I will take a crack at this and put a pr against your branch